src/corehost/common/pal.unix.cpp

```diff
@@ -75,6 +75,91 @@ bool pal::getcwd(pal::string_t* recv)
     return true;
 }
 
+namespace
+{
+    bool get_loaded_library_from_proc_maps(const pal::char_t *library_name, pal::dll_t *dll, pal::string_t *path)
+    {
+        char *line = nullptr;
+        size_t lineLen = 0;
+        ssize_t read;
+        FILE *file = pal::file_open(_X("/proc/self/maps"), _X("r"));
+        if (file == nullptr)
+            return false;
+
+        // Read maps file line by line to check fo the library
+        bool found = false;
+        pal::string_t path_local;
+        while ((read = getline(&line, &lineLen, file)) != -1)
+        {
+            char buf[PATH_MAX];
+            if (sscanf(line, "%*p-%*p %*[-rwxsp] %*p %*[:0-9a-f] %*d %s\n", buf) == 1)
+            {
+                path_local = buf;
+                size_t pos = path_local.find(DIR_SEPARATOR);
```

Comment:
Would be better to look for the last `DIR_SEPARATOR`. That way we avoid issues of finding a path to wrong library which just looks similar- for example `/../myhostfxr.sotest/lib.so`. Even if there would be a `hostfxr` later on in the list we would find the wrong library.
I know it's not very likely, but searching for the last directory separator should fix most of these cases.